lib/index.js

```diff
@@ -717,7 +717,7 @@ exports.merge = (schemas, options = {}) => {
 		if (mergedSchemas.anyOf && mergedSchemas.anyOf.length === 1) {
 			const [ anyOf ] = mergedSchemas.anyOf
 
-			return merge(omit(mergedSchemas, 'anyOf'), anyOf)
+			return combineAnyOf(omit(mergedSchemas, 'anyOf'), anyOf)
```

Comment:
I may be misusing `combineAnyOf()` here as it is meant to merge `anyOf` branches, but I believe the desired end result is the same in this case.